airbyte-workers/src/main/java/io/airbyte/workers/RecordSchemaValidator.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.airbyte.config.StandardSyncInput;
+import io.airbyte.protocol.models.AirbyteRecordMessage;
+import io.airbyte.validation.json.JsonSchemaValidator;
+import io.airbyte.validation.json.JsonValidationException;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Validates that AirbyteRecordMessage data conforms to the JSON schema defined by the source's
+ * configured catalog
+ */
+
+public class RecordSchemaValidator {
+
+  private final Map<String, JsonNode> streams;
+
+  public RecordSchemaValidator(final StandardSyncInput syncInput, final Map<String, JsonNode> streamNamesToSchemas) {
```

Comment:
Since this `syncInput` argument is no longer used by the constructor, it can be removed entirely